plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/MavenUtils.java

```diff
@@ -63,12 +81,59 @@ public static boolean hasMavenNature(IProject project) {
     }
   }
 
-  public static Artifact resolveArtifact(IProgressMonitor monitor, String groupId,
-      String artifactId, String type, String version, String classifier,
-      List<ArtifactRepository> repositories) throws CoreException {
-    Artifact artifact = MavenPlugin.getMaven().resolve(groupId, artifactId, version, type,
-        classifier, repositories, monitor);
-    return artifact;
+  public static Artifact resolveArtifact(
+      String groupId,
+      String artifactId,
+      String type,
+      String version,
+      String classifier,
+      List<ArtifactRepository> repositories,
+      IProgressMonitor monitor)
+      throws CoreException {
+    return runOperation(
+        monitor,
+        (context, system, progress) -> {
+          Artifact artifact =
+              MavenPlugin.getMaven()
+                  .resolve(groupId, artifactId, version, type, classifier, repositories, progress);
```

Comment:
Q: so previously we were calling `.resolve()` without `MavenPlugin.getMaven().createExecutionContext().execute()`. Was it not really right? (We don't use `context` here actually.)

Reply:
It was ok.  I don't think creating the context hurts.

Reply:
Looking into the implementation, `MavenImpl#resolve()` actually does execute its operation within a context, and so it should pick up the one created here.  So it doesn't hurt for us to create a context.